pkg/reconciler/pipelinerun/resources/resultrefresolution.go

```diff
@@ -130,15 +130,16 @@ func resolveResultRef(pipelineState PipelineRunState, resultRef *v1beta1.ResultR
 	var resultValue v1beta1.ResultValue
 	var err error
 	if referencedPipelineTask.IsCustomTask() {
-		runName = referencedPipelineTask.RunObject.GetObjectMeta().GetName()
-		runValue, err = findRunResultForParam(referencedPipelineTask.RunObject, resultRef)
+		runName = referencedPipelineTask.RunObjects[0].GetObjectMeta().GetName()
+		runValue, err = findRunResultForParam(referencedPipelineTask.RunObjects[0], resultRef)
```

Comment:
I'm not sure I understand what's going on here. Is this resolving results produced by a matrixed pipeline task?
Since we don't support this yet, I think it would be better to special-case matrix tasks rather than choosing `RunObjects[0]`, which is less clear. Also, is it possible `RunObjects` could be empty here, causing a reconciler panic?

Reply:
Updated to add a check for referencing results produced by a matrix and updated to looping over the list of TaskRuns/RunObjects.